website/source/docs/job-specification/affinity.html.md

```diff
@@ -0,0 +1,253 @@
+  or any [Nomad interpolated
+  values](/docs/runtime/interpolation.html#interpreted_node_vars).
+
+- `weight` `(integer:0)` - Specifies a weight for the affinity. The weight is used
+  during scoring and must be an integer between -100 to 100. Negative weights act as
+  anti affinities, causing nodes that match them to be scored lower. Weights can be used
+  when there is more than one affinity to express relative preference across them.
+
+### `operator` Values
+
+This section details the specific values for the "operator" parameter in the
+Nomad job specification for affinities. The operator is always specified as a
+string, but the string can take on different values which change the behavior of
+the overall affinity evaluation.
+
+```hcl
+affinity {
+  operator = "..."
+}
+```
+
+- `"regexp"` - Specifies a regular expression affinity against the attribute.
+  The syntax of the regular expressions accepted is the same general syntax used
+  by Perl, Python, and many other languages. More precisely, it is the syntax
+  accepted by RE2 and described at in the [Google RE2
+  syntax](https://golang.org/s/re2syntax).
+
+    ```hcl
+    affinity {
+      attribute = "..."
+      operator  = "regexp"
+      value     = "[a-z0-9]"
+      weight    = 50
+    }
+    ```
+
+- `"set_contains_all"` - Specifies a contains affinity against the attribute. The
+  attribute and the list being checked are split using commas. This will check
+  that the given attribute contains **all** of the specified elements.
+
+    ```hcl
+    affinity {
+      attribute = "..."
+      operator  = "set_contains"
+      value     = "a,b,c"
+      weight    = 50
+    }
+    ```
+- `"set_contains"`  - Same as `set_contains_all`
+
+- `"set_contains_any"` - Specifies a contains affinity against the attribute. The
+  attribute and the list being checked are split using commas. This will check
+  that the given attribute contains **any** of the specified elements.
+
+    ```hcl
+    affinity {
+      attribute = "..."
+      operator  = "set_contains"
+      value     = "a,b,c"
+      weight    = 50
+    }
+    ```
+
+- `"version"` - Specifies a version affinity against the attribute. This
+  supports a comma-separated list of values, including the pessimistic
+  operator. For more examples please see the [go-version
+  repository](https://github.com/hashicorp/go-version) for more specific
+  examples.
+
+    ```hcl
+    affinity {
+      attribute = "..."
+      operator  = "version"
+      value     = ">= 0.1.0, < 0.2"
+      weight    = 50
+    }
+    ```
+
+## `affinity` Examples
+
+The following examples only show the `affinity` stanzas. Remember that the
+`affinity` stanza is only valid in the placements listed above.
+
+### Kernel Data
+
+This example adds a preference for running on nodes which have a kernel version
+higher than "3.19".
+
+```hcl
+affinity{
+  attribute = "${attr.kernel.version}"
+  operator  = "version"
+  value     = "> 3.19"
```

Comment:
is this correct? should the operator be `>` and the value be `3.19`?

Reply:
Its valid syntax https://www.nomadproject.io/docs/job-specification/constraint.html#quot-version-quot-, we use this library that handles custom parsing https://github.com/hashicorp/go-version